js/app.js

```diff
@@ -52,31 +56,64 @@ function getRandomIndexes(quantity, min, max) {
   return uniqueIndexes;
 }
 
-//  FINISH FUNCTION TO TEST WHETHER AN OPTINO HAS BEEN PREVIOUSLY PRESENTED
-function testChoiceArray(indexArr) {
+//  tests whether a new index array can be used
+function indexArrIsUnique(indexArr) {
+  // return true if there are no previous guesses
+  if (prevChoices.length === 0) {
+    return true;
+  }
+
+  // check that index is unique to all previous choices (permutations and combinations)
+  for (var choice of prevChoices) {
+    var duplicates = 0;
+    for (var indexValue of indexArr) {
+      if (choice.includes(indexValue)) {
+        duplicates++;
+      }
+      if (duplicates === indexArr.length) {
+        return false;
+      }
+    }
+  }
+
+  //check that new selection does not have any duplicate images from the last selection
+  var lastChoiceIndex = prevChoices.length - 1;
+  var lastChoice = prevChoices[lastChoiceIndex];
+  duplicates = 0;
+  for (indexValue of indexArr) {
+    if (lastChoice.includes(indexValue)) {
+      return false;
+    }
+  }
   return true;
 }
 
+// clears all of the items from the screen before rendering
+function clearItems () {
+  while (itemsList.hasChildNodes()) {
+    itemsList.removeChild(itemsList.lastChild);
+  }
+}
+
+// renders a set of items on the page. the # of items is a parameter (3 is recommended)
 function renderItems (quantity) {
   // create new array of random indexes
   var indexArr = getRandomIndexes(quantity, 0, items.length - 1);
   // ensure that it has not been previously shown
-  if (!testChoiceArray(indexArr)) {
+  if (!indexArrIsUnique(indexArr)) {
```

Comment:
This isn't quite doing what you want, and there are still duplicates sometimes. You probably want to include a loop to keep generating a new `indexArr` while it's not unique.